README.md

```diff
@@ -1,70 +1,118 @@
 # MuSig2 Implementation with libsecp256k1
+
 This is a MuSig2 implementation using the libsecp256k1 library for EC operations.
 
 To install the library, follow the directions in [libsecp256k1](https://github.com/bitcoin-core/secp256k1).
-The library provides an optimized C library for ECDSA signatures and secret and public key operations on curve 
-secp256k1, as well as usage examples including ECDSA and Schnorr signatures.
+The library provides an optimized C library for ECDSA signatures and secret and public key operations on curve ecp256k1, as well as usage examples including ECDSA and Schnorr signatures.
 
-This implementation requires to configure the libsecp256k1 with an additional flag 
-`--enable-module-schnorrsig` as stated in [libsecp256k1](https://github.com/bitcoin-core/secp256k1).
+This implementation requires configuring the libsecp256k1 with an additional flag `--enable-module-schnorrsig` as stated in [libsecp256k1](https://github.com/bitcoin-core/secp256k1).
 
-Test the implementation with musig2test.c
+Run the example with examplemusig2.c
 
+```shell
+make ctest
+./ctest_run/ctest
 ```
-make
-./musig2test
+
+Run the Google tests with
+
+```shell
+make gtest
+./gtest_run/gtest
 ```
 
-## Verify MuSig2 with `secp256k1_schnorrsig_verify`
-The standard for 64-byte Schnorr signatures over secp256k1 uses x only encoding for _R_ and PK (aggregated public 
-key _X_ in our case) that results in 32-byte public keys and 64-byte signatures.
+Running Valgrind in MacOs can be quite painful. 
+We included a Dockerfile to run valgrind checks on MacOs with an arm chip (e.g. M1).
+To test the code with valgrind, run the following:
+```shell
+ docker build -t "valgrind:1.0" .
+ docker run -it -v $PWD:/tmp -w /tmp valgrind:1.0
+```
 
-Every valid x coordinate has two possible y coordinates, in order to avoid ambiguity the library uses points with 
-even y coordinates. The signature generation and verification are done with using only x coordinates of public key 
-and R since they only have even y coordinates.
+and once you are interacting with the container, run
+```shell
+make valgrind
+```
 
-However, to obtain _R_ and aggregated public key _X_ with even y coordinates may not be the case for every trial of 
-MuSig2.
+## Schnorr Signature of libsecp256k1
 
-The aggregated public key is `X = X_1 * a_1 + ... + X_n * a_n` and _R_ is `R = R_1 * b^(j-1) + ... + R_V * b^(V-1)`.
+The library offers an optional module implementing Schnorr signatures over the curve `secp256k1`. 
+The scheme is compatible with [BIP-340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki) that produces 64-byte Schnorr signatures. 
+BIP-340 standardized the scheme with some modifications for practical purposes.
 
-The verification of MuSig2 checks whether:
 
-```
-    G * sig = R + X * c.
-```
+### Modifications
```

Comment:
I would probably add here a small paragraph saying what are the modifications to. Maybe refer to as the 'traditional' schnorr scheme, and motivate why these motivations are of interest (I guess that BIP-340 will have their motivation).